web-marketplace/src/components/organisms/LocationForm/LocationForm.schema.ts

```diff
@@ -0,0 +1,18 @@
+import { GeocodeFeature } from '@mapbox/mapbox-sdk/services/geocoding';
+import { z } from 'zod';
+
+import { requiredMessage } from 'web-components/lib/components/inputs/validation';
+
+export const locationFormSchema = z.object({
+  'schema:location': z
+    .union([z.string(), z.custom<GeocodeFeature>()])
+    .refine(value => !!value, {
+      message: requiredMessage,
+    }),
+});
+
+export type LocationFormSchemaType = z.infer<typeof locationFormSchema>;
+
+export type SimplifiedLocationFormSchemaType = {
```

Comment:
Add to create a second type here because the initial value can only be of type `GeocodeFeature`.